PF/PF-Assignment2/FixedProductPromotion.java

```diff
@@ -0,0 +1,151 @@
+import java.util.HashMap;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.text.ParseException;
+
+public class FixedProductPromotion extends ShoppingCart implements Promotion {
+	
+	private HashMap<Long, Double> fixedDiscount;
+	
+	FixedProductPromotion(){
+		fixedDiscount = new HashMap<>();
+	}
+	
+	/*
+	 * @param   item id (long), promo code (string)
+	 * @return  minimum item price
+	 */
+	public double getMinimumPrice(long itemID, String code) {
+		
+		this.setMinimumPrice(itemID, code);
+		
+		return this.getPrice(itemID);		
+	}
+	
+	/*
+	 * @param  item id (long), promo code (String)
+	 * @return nothing
+	 * set minimum price for item after apply promo code discount and fixed discount
+	 */
+	public void setMinimumPrice(long itemID, String code) {
+		double priceAfterFixedDiscount ;
+		double price = this.getPrice(itemID);
+		
+		priceAfterFixedDiscount = price - (price * (this.fixedDiscount.get(itemID)/100.0));
+		
+		if (this.isPromotionApplicable(itemID, code)) {
+			
+			double priceAfterPromoDiscount ;
+			priceAfterPromoDiscount = price - (price *(this.getPromoDiscount(code)/100.0));
+			
+			if (priceAfterFixedDiscount < priceAfterPromoDiscount) {
+				this.setPrice(itemID, priceAfterFixedDiscount);
+			}
+			else {
+				this.setPrice(itemID, priceAfterPromoDiscount);
+			}
+		}
+	}
+	
+	/*
+	 * @param   Promotion code 
+	 * @return  discount on given promotion code
+	 */
+	public double getPromoDiscount(String code) {
+		
+		final double noDiscount = 0.0;
+		HashMap<String, Double> discount = new HashMap<>();
+		discount.put("sky10", 15.00);
+		discount.put("algo10", 10.00);
+		discount.put("laptop10", 25.00);
+		discount.put("samsung10", 40.00);
+	
+		if(discount.containsKey(code)) {
+			return discount.get(code);
+		}
+		else {
+			return noDiscount;
+		}
+		
+	}
+	
+	/*
+	 * @param   item id (int)
+	 * @return  fixed Discount on item
+	 */
+	public double getFixedDiscount(long itemID) {
+		
+		double noDiscount = 0.0;
```

Comment:
No need to store a variable if you are using it once only in that block, you can directly return it's value.